topic_1/Transaction.java

```diff
@@ -0,0 +1,27 @@
+package topic_1;
+
+public class Transaction {
+	
+	private Payment payment;
+	private float totalPrice;
+	private static int idTransaction = 0;
+	
+	public Transaction(Payment payment, float totalPrice) {
+		idTransaction++;
```

Comment:
You need to create a thread safe incremental id generator in order to provide unique Id to Payment transactions.
http://www.programcreek.com/2014/02/how-to-make-a-method-thread-safe-in-java/